Automattic/jetpack
# Nav-unification: API/WP-Admin behaviour inconsistency

Fixes https://github.com/Automattic/wp-calypso/issues/56340
Alternative for https://github.com/Automattic/jetpack/pull/21881 
#### Changes proposed in this Pull Request:

* The API endpoint used in Calypso did not had the same behaviour as the logic from WP-Admin core menu that computes the menu links, thus breaking the URL submenus in Calypso for plugins that relied on this behaviour.

#### Jetpack product discussion


N/A
#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Get an Atomic site and use Jetpack Beta and switch to this branch.
* Install the WPML plugin: 175618-pb
* Go to Calypso and make sure that the WPML menu (and submenus) are not broken.
* Since this changes affects the behaviour for all submenus, make a quick smoke test to be sure that the existing menus weren't affected.
